caddytest/caddytest_assert.go

```diff
@@ -0,0 +1,116 @@
+package caddytest
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/aryann/difflib"
+	"github.com/stretchr/testify/require"
```

Comment:
I just noticed testify is in Caddy's go.mod. The only other usage of it is for the function `assert.JSONEq`, which isn't complex (merely marshals into interface and deep-compares). We can probably get rid of it if the rest of the team agrees.

Reply:
yeah i used it because it was already in the go.mod.

fine with removing it if you guys dont like it

Reply:
I'm in favor of slimming down our dependencies, especially if it means a change in one other place.